<a href="https://colab.research.google.com/github/rddelarosa/APM1111/blob/main/SA1_Stats.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Questions

## I. (50 points)

1. (20 points). Provide descriptive statistical summaries of the entire data using JASP, Python, or R.  
2. (20 points). Provide descriptive statistical summaries of the data for each gender category using JASP, Python, or R.  
3. (10 points). Make a report based on the statistical summaries, including the results for both the combined data (not split by gender) and the gender-specific categories.  

In [19]:
import pandas as pd
import numpy as np
from scipy.stats import skew, kurtosis

# Data (50 males and 50 females)
males = [
    12,4,11,13,11,
    7,9,10,10,7,
    7,12,6,9,15,
    10,11,12,7,8,
    8,9,11,10,9,
    10,9,9,7,9,
    11,7,10,10,11,
    9,12,12,8,13,
    9,10,8,11,10,
    13,13,9,10,13
]
females = [
    11,9,7,10,9,
    10,10,7,9,10,
    11,8,9,6,11,
    10,7,9,12,14,
    11,12,12,8,12,
    12,9,10,11,7,
    12,7,9,8,11,
    10,8,13,8,10,
    9,9,9,11,9,
    9,8,9,12,11
]

# Combine into DataFrame
df = pd.DataFrame({
    "Gender": ["Male"] * len(males) + ["Female"] * len(females),
    "Hours": males + females
})

In [20]:


# Combine into DataFrame
df = pd.DataFrame({
    "Gender": ["Male"] * len(males) + ["Female"] * len(females),
    "Hours": males + females
})

# 1. Overall descriptive statistics
overall_stats = df["Hours"].describe()
overall_skew = skew(df["Hours"])
overall_kurtosis = kurtosis(df["Hours"])

print("=== Overall Descriptive Statistics ===")
print(overall_stats)
print(f"Skewness: {overall_skew:.3f}")
print(f"Kurtosis: {overall_kurtosis:.3f}")
print()

# 2. Descriptive stats by gender
grouped = df.groupby("Gender")["Hours"].agg(["mean", "median", "std", "min", "max"])
grouped["skewness"] = df.groupby("Gender")["Hours"].apply(skew)
grouped["kurtosis"] = df.groupby("Gender")["Hours"].apply(kurtosis)

print("=== Descriptive Statistics by Gender ===")
print(grouped)
print()


=== Overall Descriptive Statistics ===
count    100.000000
mean       9.760000
std        1.964945
min        4.000000
25%        9.000000
50%       10.000000
75%       11.000000
max       15.000000
Name: Hours, dtype: float64
Skewness: -0.007
Kurtosis: -0.044

=== Descriptive Statistics by Gender ===
        mean  median       std  min  max  skewness  kurtosis
Gender                                                      
Female  9.70     9.5  1.775686    6   14  0.130345 -0.516480
Male    9.82    10.0  2.154161    4   15 -0.111207  0.042378



The dataset represents the weekly hours spent talking on cell phones by 50 male and 50 female students at Midwestern University. Descriptive statistics were computed using Python to summarize the distribution of hours for the combined sample and for each gender separately.

The combined data indicate that students spend an average of 9.76 hours per week on phone calls. The distribution is nearly perfectly symmetric (skew ≈ 0) and close to normal (kurtosis ≈ 0). There is a moderate spread (SD ≈ 2 hours), with most students reporting between 9 and 11 hours per week.

Male students have an average of 9.82 hours per week, slightly above the combined average. Their data show a small negative skew (slightly more high-end users) but remain approximately normal. The higher standard deviation suggests greater variability among males in weekly phone usage.


Female students have an average of 9.7 hours per week, which is very close to the male average. Their distribution is slightly positively skewed, indicating a small number of higher phone usage cases, but overall it’s close to normal. Females show less variation than males (lower SD), suggesting more consistent phone habits.

Based on descriptive statistics, there is no substantial difference in the average time males and females spend talking on cell phones. Both groups average close to 10 hours per week, with males showing slightly more variation. The overall dataset is balanced, symmetric, and normally distributed, suggesting that gender does not play a major role in weekly phone call duration among these students.

## II. (30 points)

1. Find the (a) first, (b) second, (c) third, and (d) fourth moments of the set 2, 3, 7, 8, 10.  
2. Find the (a) first, (b) second, (c) third, and (d) fourth moments **about the mean** of the set 2, 3, 7, 8, 10.  
3. Verify:  
   \[
   m_4 = m'_4 - 4m'_1m'_3 + 6{m'_1}^2m'_2 - 3{m'_1}^4
   \]  
   for the set 2, 3, 7, 8, 10.


In [22]:
data = np.array([2, 3, 7, 8, 10])

# (a) Raw moments about origin
m1_prime = np.mean(data)
m2_prime = np.mean(data**2)
m3_prime = np.mean(data**3)
m4_prime = np.mean(data**4)

print("=== Moments about origin (m') ===")
print(f"m1' = {m1_prime:.4f}")
print(f"m2' = {m2_prime:.4f}")
print(f"m3' = {m3_prime:.4f}")
print(f"m4' = {m4_prime:.4f}")
print()

# (b) Moments about the mean
mean = m1_prime
m2 = np.mean((data - mean)**2)
m3 = np.mean((data - mean)**3)
m4 = np.mean((data - mean)**4)

print("=== Moments about the mean ===")
print(f"m2 = {m2:.4f}")
print(f"m3 = {m3:.4f}")
print(f"m4 = {m4:.4f}")
print()

# 3. Verification formula
m4_check = m4_prime - 4*m1_prime*m3_prime + 6*(m1_prime**2)*m2_prime - 3*(m1_prime**4)
print(f"Verification: m4 (from formula) = {m4_check:.4f}")
print()


=== Moments about origin (m') ===
m1' = 6.0000
m2' = 45.2000
m3' = 378.0000
m4' = 3318.8000

=== Moments about the mean ===
m2 = 9.2000
m3 = -3.6000
m4 = 122.0000

Verification: m4 (from formula) = 122.0000



## III. (20 points)

Prove that  
\[
m'_4 = m_4 + 4hm_3 + 6h^2m_2 + h^4
\]  
where \( h = m'_1 \).


In [27]:
h = m1_prime
m4_prime_check = m4 + 4*h*m3 + 6*(h**2)*m2 + h**4
print("=== Check of m4' = m4 + 4hm3 + 6h²m2 + h⁴ ===")
print(f"m4' (computed) = {m4_prime:.4f}")
print(f"m4' (formula)  = {m4_prime_check:.4f}")


=== Check of m4' = m4 + 4hm3 + 6h²m2 + h⁴ ===
m4' (computed) = 3318.8000
m4' (formula)  = 3318.8000
